### PR TITLE
Hide soft keyboard on form submission. Submit form with done button in keyboard

### PR DESCRIPTION
**Pull request by vishwesh3**
Fix:#474

### CODE REVIEW

File: src/start/PasswordAuthView.js [hunk 1]
```diff
@@ -83,6 +85,7 @@ class PasswordAuthView extends React.Component {
           value={password}
           onChangeText={newPassword => this.setState({ password: newPassword })}
           blurOnSubmit={false}
+          onSubmitEditing={this.validateForm}
```
Comment (borisyankov) [inline src/start/PasswordAuthView.js]: Great!

File: src/start/RealmScreen.js [hunk 2]
```diff
@@ -80,6 +82,7 @@ class RealmScreen extends React.Component {
               defaultValue={realm}
               onChangeText={value => this.setState({ realm: value })}
               blurOnSubmit={false}
+              onSubmitEditing={this.tryRealm}
```
Comment (borisyankov) [inline src/start/RealmScreen.js]: Good.

File: src/start/PasswordAuthView.js [hunk 3]
```diff
@@ -50,6 +50,8 @@ class PasswordAuthView extends React.Component {
   };
 
   validateForm = () => {
+    Keyboard.dismiss();
```
Comment (borisyankov) [inline src/start/PasswordAuthView.js]: We can do this better.
We would want to rework the screen itself to fit in the space above the keyboard.
Can you rework, please?

**Comment by vishwesh3** [comment]
Done.
Now content moves up when the keyboard pops up (error is shown above it). And the user doesn't need to tap on the field to set its focus.
ezgif com-video-to-gif 22

**Comment by vishwesh3** [comment]
@zulipbot Thanks for the reminder. Updated PR 👍

**Comment by borisyankov** [comment]
Looks good now.
Thanks. Merging.

**Pull request merged**